src/openApi/openApiWrapper.ts

```diff
@@ -50,6 +57,10 @@ export default class OpenApiWrapper {
 
           this.clearContext();
 
+          if (response instanceof ApiResponse) {
+            return response.withCorrelationId(this.correlationId);
```

Comment:
This is kinda hard. The response can be any object as we export the JS OpenAPI and that doesn't enforce types in the controller hooks. The worst thing that can happen now is that the response headers won't include correlation ID which is not greaaaat. 

One way to solve this is to separate the interface of the `ApiResponse` class and return that in this middleware. That way we could add/override the headers on ln 53/64

Reply:
I don't understand the proposed solution. Is there a way to enforce what ends up in response middleware? I guess all controllers would need to implement some interface that won't allow them to return anything they want?

Still, one can write a controller without implementing the interface.

Reply:
Ah yea I didn't explain it well at all. The only blocker there was that `response.headers` is `public readonly` in the class and in the interface it would be just `public`. But I removed the `readonly` from the class so that we can override it in the case when `response` is not instanceof  `ApiResponse`. It's not the cleanest solution but can't really think of anything better. Have a look pls

Reply:
To answer your point though - the issue is that we export the original `OpenApi` class that's not typed. We could and probably should type it here and return that instead. Then you would have to write controllers that return `ApiResponse`. I could even do it now. What do you think?

Reply:
One would need to write controllers that return ApiResponse because the "router" of OpenAPI will only accept such route? Daan tried that already in the DefinitelyTyped project (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/openapi-factory/index.d.ts)

Can be installed from here https://www.npmjs.com/package/@types/openapi-factory
But it most likely doesn't match what we need.

On the decision to type it or not, up to you and the time you like to devote to this :)

Reply:
Cool! It wasn't much work. I copied some stuff from the definitely typed project & added/adapted it to our types